src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -159,6 +162,7 @@ public IReadOnlyNetworkPeerCollection ConnectedPeers
             this.PeerConnectors = peerConnectors;
             this.peerDiscovery = peerDiscovery;
             this.ConnectionSettings = connectionSettings;
+            this.connectedPeersHelper = new ConnectedPeersHelper(this.loggerFactory);
```

Comment:
If we don't inject it to the factory (see my other comment then here it can definitely be injected.

Reply:
Hmm as this is a utility class then perhaps no need to inject it but it could be an internal citizen of the factory.

Reply:
Yes, that's also an option. But that way we will need to make the factory IDisposable

Reply:
Ah its not IDisposable I see, good point.
This feel more like a factory type then a connection manager type so we need to think about it.

Reply:
Each component has to have its own instance of this helper, because when the component is being destroyed it only wants to destroy those peers that it owns, not all peers.